app/src/main/java/io/apicurio/registry/rules/RulesServiceImpl.java

```diff
@@ -117,8 +131,22 @@ public void applyRule(String artifactId, long artifactVersion, ArtifactType arti
             throws RuleViolationException {
         StoredArtifact versionContent = storage.getArtifactVersion(artifactId, artifactVersion);
         // Get the rules for this artifact
-        for (RuleType ruleType : storage.getGlobalRules()) {
-            RuleConfigurationDto configurationDto = storage.getGlobalRule(ruleType);
+
+        List<RuleType> globalRules = storage.getGlobalRules();
+        List<RuleType> defaultGlobalRules = rulesProperties.getDefaultGlobalRules(globalRules);
+        if (!defaultGlobalRules.isEmpty()) {
+            // We have default global rules as well as rules from storage, so use the concatenated list
+            globalRules = Stream.concat(
+                globalRules.stream(),
+                defaultGlobalRules.stream()).collect(Collectors.toList());
+        }
+        for (RuleType ruleType : globalRules) {
+            RuleConfigurationDto configurationDto;
+            if (defaultGlobalRules.contains(ruleType)) {
+                configurationDto = rulesProperties.getDefaultGlobalRuleConfiguration(ruleType);
+            } else {
+                configurationDto = storage.getGlobalRule(ruleType);
+            }

```

Comment:
Opportunity to share logic with the other `applyRules` impl?

Reply:
Good idea - have done some refactoring.